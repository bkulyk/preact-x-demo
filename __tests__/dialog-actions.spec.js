import React from 'react';
import DialogActions from '@bitchin/dialog-actions';
import { shallow } from './enzyme';
import { expect } from './jest';

it('renders correctly with array actions', () => {
  const tree = shallow((
    <DialogActions
      actions={['cancel', 'ok']}
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with hashmap actions', () => {
  const tree = shallow((
    <DialogActions
      actions={{ cancel: 'Cancel', ok: 'OK' }}
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with hashmap actions and a default action', () => {
  const tree = shallow((
    <DialogActions
      title="test dialog"
      actions={{ cancel: 'Cancel', ok: 'OK' }}
      defaultAction="ok"
    />
  ));

  expect(tree).toMatchSnapshot();
});
