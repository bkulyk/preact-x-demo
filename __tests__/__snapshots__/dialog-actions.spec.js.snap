// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with array actions 1`] = `
<Fragment>
  <button
    className="mdc-button mdc-dialog__button"
    data-mdc-dialog-action="cancel"
    key="cancel"
    type="button"
  >
    cancel
  </button>
  <button
    className="mdc-button mdc-dialog__button"
    data-mdc-dialog-action="ok"
    key="ok"
    type="button"
  >
    ok
  </button>
</Fragment>
`;

exports[`renders correctly with hashmap actions 1`] = `
<Fragment>
  <button
    className="mdc-button mdc-dialog__button"
    data-mdc-dialog-action="cancel"
    key="cancel"
    type="button"
  >
    Cancel
  </button>
  <button
    className="mdc-button mdc-dialog__button"
    data-mdc-dialog-action="ok"
    key="ok"
    type="button"
  >
    OK
  </button>
</Fragment>
`;

exports[`renders correctly with hashmap actions and a default action 1`] = `
<Fragment>
  <button
    className="mdc-button mdc-dialog__button"
    data-mdc-dialog-action="cancel"
    key="cancel"
    type="button"
  >
    Cancel
  </button>
  <button
    className="mdc-button mdc-dialog__button"
    data-mdc-dialog-action="ok"
    data-mdc-dialog-button-default={true}
    data-mdc-dialog-initial-focus={true}
    key="ok"
    type="button"
  >
    OK
  </button>
</Fragment>
`;
